linters.go

```diff
@@ -1,30 +1,44 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
 	"os"
 	"os/exec"
 	"regexp"
+	"sort"
 	"strings"
 
 	"gopkg.in/alecthomas/kingpin.v3-unstable"
 )
 
+type LinterConfig struct {
+	Name              string
+	Command           string
+	Pattern           string
+	InstallFrom       string
+	PartitionStrategy partitionStrategy
+	IsSlow            bool
```

Comment:
Good linters are slow, there's not a lot you can do accurately with just the AST it seems. @mvdan started some work to provide interfaces to share work here https://github.com/mvdan/lint - and dominikh's megacheck helps to reuse the go-tools collection's work.